compiler/rustc_session/src/errors.rs

```diff
@@ -372,44 +378,41 @@ pub fn report_lit_error(sess: &ParseSess, err: LitError, lit: token::Lit, span:
         valid.then(|| format!("0{}{}", base_char.to_ascii_lowercase(), &suffix[1..]))
     }
 
-    let token::Lit { kind, symbol, suffix, .. } = lit;
+    let token::Lit { kind, symbol, suffix } = lit;
     let dcx = &sess.dcx;
     match err {
         LitError::InvalidSuffix => {
-            if let Some(suffix) = suffix {
-                dcx.emit_err(InvalidLiteralSuffix { span, kind: kind.descr(), suffix });
-            }
+            let suffix = suffix.unwrap();
```

Comment:
Instead of adding this `unwrap` here, could you instead change `InvalidSuffix` to `InvalidSuffix(Symbol)` holding said suffix. This won't increase the size of `LitError` since `Symbol` is a `u32` and some error variants already take an `u32`.

Reply:
The `InvalidIntSuffix` and `InvalidFloatSuffix` cases below already use `expect` to unwrap, so this is (almost) consistent with that.

Reply:
Ah, I see. Personally speaking I would update all those variants to take `Symbol` then :P. It just feels cleaner to me.

Reply:
Alright, I added another commit doing this.